### PR TITLE
[Panorama] fix get-jobs

### PR DESCRIPTION
**Pull request by jlevypaloalto**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-41030)

## Description
Fix an error in which the output the PAN-OS "show jobs all" command would contain a failed job and therefore have missing values. When the values were passed into the data class `ShowJobsAllResultData` it raised this error:
```
ShowJobsAllResultData.__init__() missing 9 required positional arguments: 'id', 'tfin', 'status', 'result', 'user', 'tenq', 'stoppable', 'positionInQ', and 'progress'
```
This PR adds default values to `ShowJobsAllResultData` to prevent this error.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 1]
```diff
@@ -9715,19 +9716,6 @@ def __post_init__(self):
         self.id = int(self.id)
 
 
-@dataclass
-class ShowJobsAllCommandResult:
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:9719]: this wasnt used before?

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 2]
```diff
@@ -11113,7 +11105,7 @@ def show_jobs(
         status=None,
         id: Optional[int] = None,
         target: Optional[str] = None
-    ) -> List[ShowJobsAllResultData]:
+    ) -> Union[list[ShowJobsAllResultData], ShowJobsAllResultData]:
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:11108]: when it isnt a list lets put into a list of one so the return type can always be a list

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 3]
```diff
@@ -6185,17 +6185,18 @@ Get all the jobs from the devices in the environment, or a single job when ID is
...
 | --- | --- | --- |
 | PANOS.JobStatus.hostid | String | Host ID. | 
-| PANOS.JobStatus.id | String | ID of job. | 
+| PANOS.JobStatus.id | String | ID of job. For failed jobs, the ID will be -1. | 
 | PANOS.JobStatus.type | String | Job type. | 
 | PANOS.JobStatus.tfin | String | Time finished. | 
 | PANOS.JobStatus.status | String | Status of the job. | 
 | PANOS.JobStatus.result | String | The result of the job. | 
-| PANOS.JobStatus.user | String | The user who initiated the job. | 
+| PANOS.JobStatus.user | Unknown | The user who initiated the job. |
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/README.md:6193]: why this change?

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 4]
```diff
@@ -6185,17 +6185,18 @@ Get all the jobs from the devices in the environment, or a single job when ID is
...
 | PANOS.JobStatus.type | String | Job type. | 
 | PANOS.JobStatus.tfin | String | Time finished. | 
 | PANOS.JobStatus.status | String | Status of the job. | 
 | PANOS.JobStatus.result | String | The result of the job. | 
-| PANOS.JobStatus.user | String | The user who initiated the job. | 
+| PANOS.JobStatus.user | Unknown | The user who initiated the job. | 
 | PANOS.JobStatus.tenq | String | The time the job was enqueued into the system. | 
 | PANOS.JobStatus.stoppable | String | Whether the job can be stopped after it started. | 
-| PANOS.JobStatus.description | String | The job description. | 
+| PANOS.JobStatus.description | Unknown | The job description. |
```
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/README.md:6196]: why this change?

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py (hunk 1 above)
Comment (jlevypaloalto) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:9719]: Exactly, it was never used

File: Packs/PAN-OS/Integrations/Panorama/README.md (hunk 3 above)
Comment (jlevypaloalto) [reply Packs/PAN-OS/Integrations/Panorama/README.md:6193]: In most cases the value returned is `null`

File: Packs/PAN-OS/Integrations/Panorama/README.md (hunk 4 above)
Comment (jlevypaloalto) [reply Packs/PAN-OS/Integrations/Panorama/README.md:6196]: In most cases the value returned is `null`

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py (hunk 2 above)
Comment (jlevypaloalto) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:11108]: Screenshot 2024-09-02 at 14 25 37

Comment (jlevypaloalto) [reply Packs/PAN-OS/Integrations/Panorama/Panorama.py:11108]: there seems to be a reason it works this way.

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 5]
```diff
@@ -6185,17 +6185,18 @@ Get all the jobs from the devices in the environment, or a single job when ID is
...
 | PANOS.JobStatus.result | String | The result of the job. | 
-| PANOS.JobStatus.user | String | The user who initiated the job. | 
+| PANOS.JobStatus.user | Unknown | The user who initiated the job. | 
 | PANOS.JobStatus.tenq | String | The time the job was enqueued into the system. | 
 | PANOS.JobStatus.stoppable | String | Whether the job can be stopped after it started. | 
-| PANOS.JobStatus.description | String | The job description. | 
+| PANOS.JobStatus.description | Unknown | The job description. | 
 | PANOS.JobStatus.positionInQ | String | The position of the job in the current job queue. | 
 | PANOS.JobStatus.progress | String | The numerical progress of the job. | 
+| PANOS.JobStatus.warnings | Unknown | The warnings of the job. |
```
[the remark this answers is not in the document]
Comment (jlevypaloalto) [reply Packs/PAN-OS/Integrations/Panorama/README.md:6199]: will revert

File: Packs/PAN-OS/Integrations/Panorama/README.md (hunk 3 above)
Comment (JudahSchwartz) [inline Packs/PAN-OS/Integrations/Panorama/README.md:6193]: still says unknown here

**Review by JudahSchwartz — APPROVED** [review]

**Comment by jlevypaloalto** [comment]
This PR needs a force-merge as it we have a flaky TPB. Issue: [CIAC-11260](https://jira-dc.paloaltonetworks.com/browse/CIAC-11260)
